### ISSUE TITLE
Excel 2019, При активной вкладке "Navferty's Tools" очень заметные фризы интерфейса!

### ISSUE THREAD

**Issue by uom42** [navferty/NavfertyExcelAddIn#83]
Excel 2019
При активной вкладке "Navferty's Tools" очень сильные фризы интерфейса при движении мышью и кликании!
Даже разворачивание / сворачивание окна тормозит 1-2 сек.

(SSD 200GB free + 128GB RAM / Ryzen-5)
Возможно из-за отслеживания истории действий?

**Comment by uom42** [comment]
Да, после удаления дополнения всё заработало быстро. Проблема точно не в офисе и не в железе.

**Comment by navferty** [comment]
Отслеживание истории происходит только непосредственно во время записи новых значений, при вызове функций обновляющих значения, так что непохоже. Скорее с отрисовкой вкладки. Наблюдаются ли проблемы при включенной надстройке, но неактивной вкладке?

Есть ли возможность прикрепить логи (в папке `%TEMP%/Navferty/logs/`)?

**Comment by navferty** [comment]
@uom42 Воспроизводится ли проблема, если переустановить надстройку? Наблюдается ли существенная нагрузка по памяти или CPU в диспетчере задач?

**Comment by uom42** [comment]
1) Сразу после запуска, пока не активна вкладка Nf'T - всё работает быстро.
2) Первый клик по вкладке Nf'T - уже происходит с тормозом в 2 сек. и потом тормоза прололжаются
3) Последующее переключение на любую другую вкладку - убирает тормоза. UI начинает работать быстро.
4) Переустановка дополнения не убирает проблему.

5) Простое движение мыши над кнопками развёрнутой ленты с активной вкладкой Nf'T - происходит с заметными тормозами.
при этом возрастает загрузка ЦП (даже ничего не нажимая).

6) На приложенном скрине - нажатие кнопки развёртывания окна на весь экран / восстановления размера окна (средняя кнопка из 3х справа сверху).
![Скриншот 2021-07-05 17 25 35](https://user-images.githubusercontent.com/6280840/124458049-bd9e8100-ddb6-11eb-87c7-42cd95f045f3.png)

**Comment by navferty** [comment]
Пока не удалось воспроизвести проблему локально.

Уточните пожалуйста версию установленной надстройки (версию установленной программы можно увидеть в списке установленных программ, в меню `Add or remove programs`).

Воспроизводится ли проблема в последней версии надстройки 1.4.0.361 (релиз от 5 июля)? В предыдущей версии (от 4 июля) могла быть проблема с отсутствие изображения для двух иконок. Вероятно, это могло быть причиной проблем с UI.

**Comment by uom42** [comment]
версия setup.exe = 16.0.28315.86, 
В add/remove = 1.4.0.361
Всего было 2 установки, между ними 1 удаление.
setup.exe запускал один и тот же, но в прошлый раз в дополнении небыло значков в "вид", а сегодня после установки они уже есть. Подозреваю, что setup при установки скачивает последнюю версию.
Папка "C:\Users\ххх\AppData\Roaming\Microsoft\AddIns" пустая, адрес надстройки указывает на vsto-файл с онлайна...

> Воспроизводится ли проблема в последней версии надстройки 1.4.0.361
Да, тормоза есть.

**Comment by navferty** [comment]
@uom42  
Была опубликована версия 1.4.1.368 в которой добавлены логи на инфраструктурные методы, используемые при рендеринге вкладки с надстройкой (`GetLabel`, `GetImage`, `GetSupertip`).

Просьба установить свежую версию (setup.exe действительно можно использовать тот же самый, он получает vsto файл с мета-информацией о надстройке вот [тут](https://navferty.ru/deploy/NavfertyExcelAddIn.vsto) ), и по возможности прикрепить свежие логи - если проблема с зависанием UI будет актуальна.

**Comment by uom42** [comment]
Переустановил, тормоза остались.
Логи прикрепляю.
[2021-07-14.log](https://github.com/navferty/NavfertyExcelAddIn/files/6813591/2021-07-14.log)

**Comment by navferty** [comment]
Наблюдаются ли в настоящее время проблемы на актуальной версии надстройки? Локально воспроизвести проблему ни на Office 365, ни в 2019-й версии так и не получилось

**Comment by uom42** [comment]
Да, всё так-же, но надстройку я не удалял и не переустанавливал с момента последнего общения.
Office 2019, с обновлённым UI (вкладки с подчёркиваеием).

**Comment by uom42** [comment]
Обновил надстройку, тормоза есть.

**Comment by navferty** [comment]
Пока, к сожалению, у меня нет версий о причинах тормозов. Могли бы Вы запустить Excel, записав трассировку через [Windows Performance Recorder](https://docs.microsoft.com/en-us/windows-hardware/test/wpt/windows-performance-analyzer) ?

Достаточно установить Windows Performance Toolkit:

![image](https://user-images.githubusercontent.com/20906347/135766227-48fc0cd8-4c47-454f-a760-ff131933fb70.png)

Могу предложить обменяться контактами, если есть возможность на звонок с расшариванием экрана, напишите в почту navferty@ymail.com или telegram: [navferty](https://t.me/navferty)
